### ISSUE TITLE
Crash when deleting a planet sometimes

### ISSUE THREAD

**Issue by smcameron** [smcameron/space-nerds-in-space#171]
Here's a stack trace:

```
Core was generated by `./snis_server -l localhost -L KARADO --enable-enscript -m narnia -s karado'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007f12f02f4c37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
56	../nptl/sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  0x00007f12f02f4c37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007f12f02f8028 in __GI_abort () at abort.c:89
#2  0x00007f12f02edbf6 in __assert_fail_base (fmt=0x7f12f0442058 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", 
    assertion=assertion@entry=0x49c2a8 "i >= 0", file=file@entry=0x49b453 "snis_server.c", line=line@entry=2766, 
    function=function@entry=0x4a7860 <__PRETTY_FUNCTION__.13636> "push_cop_mode") at assert.c:92
#3  0x00007f12f02edca2 in __GI___assert_fail (assertion=0x49c2a8 "i >= 0", file=0x49b453 "snis_server.c", line=2766, 
    function=0x4a7860 <__PRETTY_FUNCTION__.13636> "push_cop_mode") at assert.c:101
#4  0x00000000004519ec in push_cop_mode ()
#5  0x000000000047ea32 in ship_move ()
#6  0x000000000040b1b9 in main ()
```

**Comment by smcameron** [comment]
Code looks like this:

```
static void push_cop_mode(struct snis_entity *cop)
{
        int i, npoints;
        float orbit_radius;
        struct snis_entity *p;
        union quat q;
        struct ai_cop_data *patrol;
        union vec3 home_planet;

        cop->tsd.ship.nai_entries = 1;
        cop->tsd.ship.ai[0].ai_mode = AI_MODE_COP;

        i = lookup_by_id(cop->tsd.ship.home_planet);
        assert(i >= 0);
        p = &go[i];
        home_planet.v.x = p->x;
        home_planet.v.y = p->y;
        home_planet.v.z = p->z;
```

So the assert is not correct, as clearly the cop's home planet could get deleted. Maybe he should adopt a new home planet on the spot, or give up on becoming a cop.

**Comment by smcameron** [comment]
Fixed by 5aeceda34835d221b3c5536d4f95cdd883d58fa5
